libsolidity/formal/SMTEncoder.h

```diff
@@ -70,7 +70,7 @@ class SMTEncoder: public ASTConstVisitor
 
 	/// @returns the FunctionDefinition of a FunctionCall
 	/// if possible or nullptr.
-	static std::pair<FunctionDefinition const*, ContractDefinition const*> functionCallToDefinition(FunctionCall const& _funCall, ContractDefinition const* _contract = nullptr);
+	static FunctionDefinition const* functionCallToDefinition(FunctionCall const& _funCall, ContractDefinition const* _scopeContract, ContractDefinition const* _contextContract);
```

Comment:
Could you document here or the in the implementation, what is `_scopeContract` and what is the `_contextContract`? Or rather, what is the difference between them.
I am reading it as `_contextContract` is the contract currently being analyzed (like the actual type of `this` in other programming languages) and `_scopeContract` is the contract that contains the definition of currently analyzed function. So if `contract C is B` and that analysis reached some function `B.f` when analyzing contract `C` then `_contextContract` is `C` but `_scopeContract` is `B`. Is that correct?

Reply:
Correct! Will add clarification

Reply:
Added